frontend/src/components/Contexts/MissionListsContext.tsx

```diff
@@ -31,9 +39,77 @@ const fetchMissions = (params: {
 }): Promise<Mission[]> => {
     return BackendAPICaller.getMissionRuns(params).then((response) => response.content)
 }
-export const useMissions = (refreshInterval: number): MissionsResult => {
+export const useMissions = (): MissionsResult => {
     const [ongoingMissions, setOngoingMissions] = useState<Mission[]>([])
     const [missionQueue, setMissionQueue] = useState<Mission[]>([])
+    const { registerEvent, connectionReady } = useSignalRContext()
+
+    useEffect(() => {
+        if (connectionReady) {
+            registerEvent('mission run created', (username: string, message: string) => {
+                const newMission: Mission = JSON.parse(message)
+                if (missionQueue.find((m) => m.id === newMission.id))
+                    setMissionQueue((oldQueue) => [...oldQueue, newMission])
+                else
+                    setMissionQueue((oldQueue) => {
+                        let missionQueueCopy = [...oldQueue]
+                        missionQueueCopy = upsertList(missionQueueCopy, newMission)
+                        return [...missionQueueCopy]
+                    })
+            })
+            registerEvent('mission run updated', (username: string, message: string) => {
+                let updatedMission: Mission = JSON.parse(message)
+                // This conversion translates from the enum as a number to an enum as a string
+                updatedMission.status = Object.values(MissionStatus)[updatedMission.status as unknown as number]
+
+                setMissionQueue((oldQueue) => {
+                    const oldQueueCopy = [...oldQueue]
+                    const existingMissionIndex = oldQueue.findIndex((m) => m.id === updatedMission.id)
+                    if (existingMissionIndex !== -1) {
+                        if (updatedMission.status !== MissionStatus.Pending)
+                            oldQueueCopy.splice(existingMissionIndex, 1)
+                        else oldQueueCopy[existingMissionIndex] = updatedMission
```

Comment:
Can this be extracted into a method to hide the complex logic and rather have a good name for the method indicating what it does?

Reply:
I think it should be better now